### PR TITLE
a little more roubust app.sh

### PR DESCRIPTION
**Pull request by goern**
Signed-off-by: Christoph Görn <goern@redhat.com>

## Related Issues and Dependencies

put in some default for parameters, set the web host of Faust to 0.0.0.0

## This introduces a breaking change

- [ ] Yes
- [X] No

### CODE REVIEW

File: app.sh [hunk 1]
```diff
@@ -4,20 +4,32 @@
...
-if [ "$SUBCOMMAND" = "producer" ]
+set -o nounset
+set -o errexit
+set -o errtrace
+set -o pipefail
+trap 'echo "Aborting due to errexit on line $LINENO. Exit code: $?" >&2' ERR
+
+# TODO(pacospace) maybe this is a good inspiration: https://github.com/xwmx/bash-boilerplate/blob/master/bash-commands
+
+FAUST_COMMAND="faust --web-host 0.0.0.0"
```
Comment (KPostOffice) [inline app.sh:15]: Why do we set the `web-host` option?

Comment (goern) [reply app.sh:15]: So that it binds to all Interfaces and not just localhost

Comment (pacospace) [inline app.sh:15]: `--web-host 0.0.0.0` cannot be applied to faust `main` but only to  faust `worker`.
Moreover `--web-host 0.0.0.0` is not a `faust` cli parameter, but a `faust worker` cli parameter.

Comment (pacospace) [reply app.sh:15]:
```
SUBCOMMAND=${SUBCOMMAND:-consumer}
DEBUG_LEVEL=${DEBUG_LEVEL:-0}

if [ "$SUBCOMMAND" == "producer" ]
then
    if [ "$DEBUG_LEVEL" -eq 1 ]
    then
        exec faust --debug --loglevel debug -A producer main
    else
        exec faust -A producer main
    fi
elif [ "$SUBCOMMAND" = "consumer" ]
then
    if [ "$DEBUG_LEVEL" -eq 1 ]
    then
        exec faust --debug --loglevel debug -A consumer worker --web-host 0.0.0.0
    else
        exec faust -A consumer worker --web-host 0.0.0.0
    fi
fi

```

**Comment by goern** [comment]
Well, i want to prevent myself from committing unencrypted secrets ;)

File: app.sh [hunk 2]
```diff
@@ -4,20 +4,22 @@
...
+DEBUG_FLAGS=""
+
+if [ "$DEBUG_LEVEL" -eq 1 ]; then
+    DEBUG_FLAGS="--debug --loglevel debug"
+fi
+
+if [ "$SUBCOMMAND" == "producer" ]; then
+    exec faust ${DEBUG_FLAGS} -A producer main
+elif [ "$SUBCOMMAND" = "consumer" ]; then
+    exec faust ${DEBUG_FLAGS} -A consumer --web-host 0.0.0.0 worker
```
Comment (pacospace) [inline app.sh:24]: `exec faust ${DEBUG_FLAGS} -A consumer worker --web-host 0.0.0.0`

Comment (goern) [reply app.sh:24]: argh

**Pull request merged**